hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ReplicationManager.java

```diff
@@ -192,24 +193,19 @@ public void stop() {
    * ReplicationMonitor thread runnable. This wakes up at configured
    * interval and processes all the containers in the system.
    */
-  private void run() {
+  private synchronized void run() {
     try {
       while (running) {
-        try {
-          final long start = Time.monotonicNow();
-          final Set<ContainerID> containerIds =
-              containerManager.getContainerIDs();
-          containerIds.forEach(this::processContainer);
-          LOG.info("Replication Monitor Thread took {} milliseconds for" +
-                  " processing {} containers.", Time.monotonicNow() - start,
-              containerIds.size());
-          if (!Thread.interrupted()) {
-            Thread.sleep(interval);
-          }
-        } catch (InterruptedException ex) {
-          // Wakeup and process the containers.
-          LOG.debug("Replication Monitor Thread got interrupted.");
-        }
+        final long start = Time.monotonicNow();
+        final Set<ContainerID> containerIds =
+            containerManager.getContainerIDs();
+        containerIds.forEach(this::processContainer);
+
+        LOG.info("Replication Monitor Thread took {} milliseconds for" +
+                " processing {} containers.", Time.monotonicNow() - start,
+            containerIds.size());
+
+        wait(interval);
       }
     } catch (Throwable t) {
       // When we get runtime exception, we should terminate SCM.
```

Comment:
Don't call terminate if running is false. In that case, just exit the thread. See RedundancyMonitor#run in HDFS BlockManager.java.